include/veriblock/blockchain/pop/fork_resolution.hpp

```diff
@@ -364,6 +364,11 @@ struct PopAwareForkResolutionComparator {
     for (size_t i = 0, size = payloads.size(); i < size; i++) {
       auto& p = payloads[i];
 
+      // containing block must be correct (current)
+      if (p.containingBlock != index.header) {
+        return state.setIndex(i).addStackFunction("Comparator::addPayloads");
+      }
```

Comment:
Are you sure you want to drop away the entire payloads array if only one element is wrong? An alternative is to skip bad element.

Reply:
Every tree is a state machine. Single state change is made through acceptBlock. Even if a single tiny piece of single acceptBlock fails, we must cancel state change